airflow/models/baseoperator.py

```diff
@@ -196,16 +216,37 @@ def apply_defaults(self, *args: Any, **kwargs: Any) -> Any:
             self._BaseOperator__instantiated = True
             return result
 
+        apply_defaults.__non_optional_args = non_optional_args  # type: ignore
+        apply_defaults.__param_names = set(non_varaidc_params.keys())  # type: ignore
```

Comment:
```suggestion
        apply_defaults.__param_names = set(non_variadic_params)  # type: ignore
```

Reply:
And should we be able to mark __paam_names as `Set[str]` ?

Reply:
The ignore here is cos of the binding/renaming of `__vars`, not the type of it, so I don't think typing will ever help with `__var` access across class.